Fix quote escaping when serializing strings.

ghislainfourny opened the pull request:
Quotes and newlines were not escaped properly.

CanBerker commented on src/main/resources/test_files/runtime/QuoteEscaping.jq:
```diff
@@ -0,0 +1,2 @@
+(:JIQS: ShouldRun; Output="{ "foo\"foo\nbar" : "foo\"foo\nbar" }" :)
+json-file("./src/main/resources/quotes.json")
```
The test file is actually located under ./src/main/resources/**queries**/quotes.json
I'm surprised that the testing suite works with either of these address inputs. The shell application only accepts the latter as a valid parameter. Throws an input path does not exist error otherwise

ghislainfourny replied:
Very good catch. The test wasn't executed because the extension must be .iq. I have fixed this.

CanBerker commented:
I just wanted to present the following cases from the shell application for sanity checking as I'm not very familiar with the escaping mechanism of sparksoniq. The grammar file indicates that ESC is detected with "\\" characters. Are the following behavior as expected? (particularly the output of the second query)

```
jiqs$ { "foo\"foo\nbar" : "foo\"foo\nbar" }
>>>
>>>
[ERROR] Error [err: XPST0003]LINE:1:COLUMN:14:Parser failed. foonbar
jiqs$ { "foo\\"foo\\nbar" : "foo\\"foo\\nbar" }
>>>
>>>
{ "foo\\"foo\nbar" : "foo\\"foo\nbar" }
```

ghislainfourny commented:
> I just wanted to present the following cases from the shell application for sanity checking as I'm not very familiar with the escaping mechanism of sparksoniq. The grammar file indicates that ESC is detected with "\" characters. Are the following behavior as expected? (particularly the output of the second query)
> 
> ```
> jiqs$ { "foo\"foo\nbar" : "foo\"foo\nbar" }
> >>>
> >>>
> [ERROR] Error [err: XPST0003]LINE:1:COLUMN:14:Parser failed. foonbar
> jiqs$ { "foo\\"foo\\nbar" : "foo\\"foo\\nbar" }
> >>>
> >>>
> { "foo\\"foo\nbar" : "foo\\"foo\nbar" }
> ```

This is a known bug that is orthogonal to this PR, namely, string literals in a query don't support escaping yet ( see https://github.com/Sparksoniq/sparksoniq/issues/119 ). This is the reason why I used JSON input for testing my change, because there the escaping is done correctly by the JSON parser. When Issue #119 gets solved, the queries you tried will work correctly.

ghislainfourny commented:
The tests fail because of the way the expected output is parsed by the testing code. I will take a closer look.

ghislainfourny commented:
I fixed the test -- note that the test output string has always been a "hack" where the newline escape sequence was manually interpreted as a new line. I have fixed it so that \\n is indeed the actual two-character string "\n".

All of this will be fixed when we migrate our tests to a cleaner setup with the output in a separate file, exactly matching the actual output.

ghislainfourny commented:
Thanks a lot, Can!
